tests/test_processor.py

```diff
@@ -222,3 +246,19 @@ def test_add_remove_ports():
     for i in range(6):
         assert processor.get_input_port(i) is None
         assert processor.get_output_port(i) is None
+
+
+def test_phase_quantization():
+    nm = NoiseModel(phase_imprecision=0.1)
+    p0 = Processor("SLOS", catalog["mzi phase first"].build_circuit(phi_a=0.596898191919898198,
+                                                                    phi_b=0.16561561651616))
+    p1 = Processor("SLOS", catalog["mzi phase first"].build_circuit(phi_a=0.596898191919898198,
+                                                                    phi_b=0.16561561651616), noise=nm)
+    p2 = Processor("SLOS", catalog["mzi phase first"].build_circuit(phi_a=0.6,
+                                                                    phi_b=0.2))
+
+    p0.with_input(BasicState([1, 1]))
+    p1.with_input(BasicState([1, 1]))
+    p2.with_input(BasicState([1, 1]))
+    assert p0.probs()["results"] != pytest.approx(p1.probs()["results"])
+    assert p1.probs()["results"] == pytest.approx(p2.probs()["results"])
```

Comment:
This is not working
```
    p1.noise = NoiseModel()
    assert p0.probs()["results"] == pytest.approx(p1.probs()["results"])
```
This is working
```
    p0.noise = nm
    assert p0.probs()["results"] == pytest.approx(p1.probs()["results"])
```
I understand that we cannot "undo" the calculation of the phases for know but we either need to support it or to put a warning when going from phase imprecision to no phase imprecision.

Reply:
Thanks for having caught that. I've copied the circuit that's used in the simulation, so that we still know the original phase values.